### PR TITLE
fix: enforce product and team isolation

### PR DESCRIPTION
**Pull request by bradygaster**
## Summary

- add a CI-enforced three-scenario build/package comparison with `.squad/**` absent, synthetic, and live
- make multi-word member labels use one slug contract across activation, assignment, and binding validation
- classify intentional destination-only templates so stale package assets fail tests
- remove dogfood cast identities from reusable policy and model-selection examples

## Validation

- static Node syntax checks and `git diff --check` pass locally
- local dependency restore was attempted only after configuring the required proxy; the proxy lacks `eslint@10.10.0`, and direct npm access is blocked, so the locked build/test suite is delegated to GitHub CI

The live `.squad/**` tree is unchanged.

### CODE REVIEW

File: workflows/squad.md [hunk 1]
```diff
@@ -1736,8 +1741,8 @@ supported order.
 
 **Label set, per issue:**
 
-- Work item: `squad`, plus `squad:{owner}` derived from that row's own frozen
-  certified `Owner`, lowercased. `@copilot` maps to the existing `squad:copilot`
+- Work item: `squad`, plus `squad:{owner-slug}` derived from that row's own frozen
+  certified `Owner` by the slug rule above. `@copilot` maps to the existing `squad:copilot`
   routing label — never `squad:@copilot`. Re-read each row's frozen `Owner`; never
   inherit the phase issue's owner or carry the previous row's value forward.
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply workflows/squad.md]: Updated phase issue computation and reporting to use `squad:{owner-slug}` and explicitly derive it with the same label-sync slug rule used for work items and bindings.

File: scripts/verify-team-isolation.mjs [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    const source = join(ROOT, file);
+    const target = join(destination, file);
+    mkdirSync(dirname(target), { recursive: true });
+    cpSync(source, target, {
+      dereference: false,
+      preserveTimestamps: true,
+    });
+  }
+  return files.length > 0;
+}
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply scripts/verify-team-isolation.mjs:90]: Updated the isolation verifier to copy the complete existing `.squad` directory recursively, including ignored runtime inputs such as config, sessions, logs, and inbox state. The live scenario now runs whenever that directory exists instead of depending on tracked files.

File: workflows/squad.md [hunk 3]
```diff
@@ -2404,7 +2409,7 @@ Root → Epics → Tasks. Phase-specific: filter to matching phase heading.
 
 - Title: task title
 - Temporary ID: `temporary_id: "#aw_task{N}"` per the Temporary-ID Contract. Required, and unique across every epic and task in this run.
-- Labels: `squad` (0075ca), `squad:{agent}` (e4e669) where `{agent}` is **this task's own `Agent` cell**, lowercased — read from the implementation-plan row whose `#` matches this task. Map `@copilot` to `squad:copilot`. Never inherit the parent epic's agent, and never carry the previous task's value forward: re-read the `Agent` cell for every task, because consecutive tasks under one epic routinely have different agents. No `size:*` labels unless policy says so.
+- Labels: `squad` (0075ca), `squad:{agent-slug}` (e4e669) where `{agent-slug}` is derived from **this task's own `Agent` cell** using the same lowercase/non-alphanumeric/hyphen slug rule as label synchronization — read from the implementation-plan row whose `#` matches this task. Map `@copilot` to `squad:copilot`. Never inherit the parent epic's agent, and never carry the previous task's value forward: re-read the `Agent` cell for every task, because consecutive tasks under one epic routinely have different agents. No `size:*` labels unless policy says so.
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply workflows/squad.md:2416]: Updated the shared Label Pre-flight to certify the lowercased raw roster value, then derive `squad:{agent-slug}` by replacing non-alphanumeric runs with hyphens and trimming them. Provisioning, correspondence, and reporting references now use the slugged label contract consistently.

**Comment by bradygaster** [comment]
@copilot please do a review of this PR and make sure it is furthering our direction towards decoupling the team from the product and that it fails to introduce any test or build breaks or user experience issues.

File: packages/squad-sdk/templates/workflows/squad-issue-assign.yml [hunk 4]
```diff
@@ -23,6 +23,31 @@ jobs:
             const fs = require('fs');
             const issue = context.payload.issue;
             const label = context.payload.label.name;
+            const slugify = value => value
+              .toLowerCase()
+              .replace(/[^a-z0-9]+/g, '-')
+              .replace(/^-|-$/g, '');
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply packages/squad-sdk/templates/workflows/squad-issue-assign.yml]: Updated the SDK assignment template mirror to use run-based edge trimming and added executable coverage across all assignment copies.

File: packages/squad-sdk/templates/workflows/sync-squad-labels.yml [hunk 5]
```diff
@@ -153,6 +153,17 @@ jobs:
             ];
 
             function slugify(t) { return t.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, ''); }
+            function assertUniqueMemberSlugs(teamMembers) {
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply packages/squad-sdk/templates/workflows/sync-squad-labels.yml]: Updated the SDK label-sync template mirror to use run-based edge trimming and added executable coverage across all sync copies.

File: .github/workflows/squad-issue-assign.yml [hunk 6]
```diff
@@ -48,6 +48,31 @@ jobs:
             const fs = require('fs');
             const issue = context.payload.issue;
             const label = context.payload.label.name;
+            const slugify = value => value
+              .toLowerCase()
+              .replace(/[^a-z0-9]+/g, '-')
+              .replace(/^-|-$/g, '');
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply .github/workflows/squad-issue-assign.yml]: Updated the assignment slugifier to trim complete leading and trailing hyphen runs with `/^-+|-+$/g`. Added regression coverage for `---Quartz Navigator---` across every assignment workflow copy.

File: packages/squad-cli/templates/workflows/sync-squad-labels.yml [hunk 7]
```diff
@@ -153,6 +153,17 @@ jobs:
             ];
 
             function slugify(t) { return t.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, ''); }
+            function assertUniqueMemberSlugs(teamMembers) {
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply packages/squad-cli/templates/workflows/sync-squad-labels.yml]: Updated the CLI label-sync template mirror to use run-based edge trimming; mirror-wide tests cover the exact edge case.

File: .squad-templates/workflows/sync-squad-labels.yml [hunk 8]
```diff
@@ -153,6 +153,17 @@ jobs:
             ];
 
             function slugify(t) { return t.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, ''); }
+            function assertUniqueMemberSlugs(teamMembers) {
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply .squad-templates/workflows/sync-squad-labels.yml]: Updated the canonical label-sync template to use run-based edge trimming and covered it with executable slug and collision tests.

File: .squad-templates/workflows/squad-issue-assign.yml [hunk 9]
```diff
@@ -23,6 +23,31 @@ jobs:
             const fs = require('fs');
             const issue = context.payload.issue;
             const label = context.payload.label.name;
+            const slugify = value => value
+              .toLowerCase()
+              .replace(/[^a-z0-9]+/g, '-')
+              .replace(/^-|-$/g, '');
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply .squad-templates/workflows/squad-issue-assign.yml]: Updated the canonical assignment template to use run-based edge trimming and covered it with the shared executable assignment regression test.

File: packages/squad-cli/templates/workflows/squad-issue-assign.yml [hunk 10]
```diff
@@ -23,6 +23,31 @@ jobs:
             const fs = require('fs');
             const issue = context.payload.issue;
             const label = context.payload.label.name;
+            const slugify = value => value
+              .toLowerCase()
+              .replace(/[^a-z0-9]+/g, '-')
+              .replace(/^-|-$/g, '');
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply packages/squad-cli/templates/workflows/squad-issue-assign.yml]: Updated the CLI assignment template mirror to use run-based edge trimming; mirror-wide tests exercise the punctuation-edge case.

File: templates/workflows/sync-squad-labels.yml [hunk 11]
```diff
@@ -153,6 +153,17 @@ jobs:
             ];
 
             function slugify(t) { return t.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, ''); }
+            function assertUniqueMemberSlugs(teamMembers) {
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply templates/workflows/sync-squad-labels.yml]: Updated the root label-sync template mirror to use run-based edge trimming and covered it with executable slug and collision tests.

File: templates/workflows/squad-issue-assign.yml [hunk 12]
```diff
@@ -23,6 +23,31 @@ jobs:
             const fs = require('fs');
             const issue = context.payload.issue;
             const label = context.payload.label.name;
+            const slugify = value => value
+              .toLowerCase()
+              .replace(/[^a-z0-9]+/g, '-')
+              .replace(/^-|-$/g, '');
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply templates/workflows/squad-issue-assign.yml]: Updated the root assignment template mirror to use run-based edge trimming and covered it with the shared executable workflow test.

File: .github/workflows/sync-squad-labels.yml [hunk 13]
```diff
@@ -157,6 +157,17 @@ jobs:
             ];
 
             function slugify(t) { return t.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, ''); }
+            function assertUniqueMemberSlugs(teamMembers) {
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply .github/workflows/sync-squad-labels.yml]: Updated label-sync slugification to trim complete edge runs. The executable mirror test now verifies punctuation-wrapped multi-word names normalize to `quartz-navigator`.

File: scripts/check-agent-binding.mjs [hunk 14]
```diff
@@ -43,6 +43,10 @@ function normalize(value) {
   return typeof value === 'string' ? value.trim().toLowerCase() : '';
 }
 
+function slugify(value) {
+  return normalize(value).replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, '');
+}
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply scripts/check-agent-binding.mjs:48]: Updated binding validation to trim complete edge runs with `/^-+|-+$/g`. Added a binding regression test proving `---Quartz Navigator---` expects and validates `squad:quartz-navigator`.

**Review by tamirdresher — APPROVED** [review]
